tests/integration/cmd_app_test.go

```diff
@@ -27,102 +23,128 @@ var _ = Describe("odo app command tests", func() {
+			})
+
+			It("should describe specific app", func() {
+				helper.Cmd("odo", "app", "describe", appName, "--project", commonVar.Project).ShouldPass()
+			})
+
+			It("should describe specific app in JSON format", func() {
+				actualDesAppJSON := helper.Cmd("odo", "app", "describe", appName, "--project", commonVar.Project, "-o", "json").ShouldPass().Out()
+				valuesDes := gjson.GetMany(actualDesAppJSON, "kind", "metadata.name", "metadata.namespace")
+				expectedDes := []string{"Application", appName, commonVar.Project}
+				Expect(helper.GjsonMatcher(valuesDes, expectedDes)).To(Equal(true))
+			})
+
+			When("changing to context directory", func() {
+
+				BeforeEach(func() {
+					helper.Chdir(commonVar.Context)
+				})
+
+				It("should list apps", func() {
+					appListOutput := helper.Cmd("odo", "app", "list", "--project", commonVar.Project).ShouldPass().Out()
+					Expect(appListOutput).To(ContainSubstring(appName))
+				})
+
+				It("should list apps in json format", func() {
+					actualCompListJSON := helper.Cmd("odo", "list", "-o", "json", "--project", commonVar.Project).ShouldPass().Out()
+					valuesL := gjson.GetMany(actualCompListJSON, "kind", "devfileComponents.0.metadata.name", "devfileComponents.0.metadata.namespace")
+					expectedL := []string{"List", "nodejs", commonVar.Project}
+					Expect(helper.GjsonMatcher(valuesL, expectedL)).To(Equal(true))
+				})
+
+				It("should decsribe app", func() {
+					helper.Cmd("odo", "app", "describe", "--project", commonVar.Project).ShouldPass()
+					actualDesAppJSON := helper.Cmd("odo", "app", "describe", "app", "-o", "json", "--project", commonVar.Project).ShouldPass().Out()
+					valuesDes := gjson.GetMany(actualDesAppJSON, "kind", "metadata.name", "metadata.namespace")
+					expectedDes := []string{"Application", "app", commonVar.Project}
+					Expect(helper.GjsonMatcher(valuesDes, expectedDes)).To(Equal(true))
+				})
+			})
+		})
 
-	Context("When running app describe with storage added in component in directory that doesn't contain .odo config directory", func() {
-		It("should successfully execute describe", func() {
-			helper.CopyExample(filepath.Join("source", "nodejs"), commonVar.Context)
-			helper.Cmd("odo", "component", "create", "--s2i", "nodejs", cmpName, "--app", appName, "--project", commonVar.Project, "--context", commonVar.Context).ShouldPass()
-			helper.Cmd("odo", "storage", "create", "storage-one", "--context", commonVar.Context, "--path", mountPath, "--size", size).ShouldPass()
-			helper.Cmd("odo", "push", "--context", commonVar.Context).ShouldPass()
-			helper.Cmd("odo", "app", "describe", appName, "--project", commonVar.Project).ShouldPass()
+		When("adding storage", func() {
 
-		})
+			mountPath := "/data"
+			size := "1Gi"
+
+			BeforeEach(func() {
+				helper.Cmd("odo", "storage", "create", "storage-one", "--context", commonVar.Context, "--path", mountPath, "--size", size).ShouldPass()
+			})
 
+			When("odo push is executed", func() {
```

Comment:
@feloy Can we merge this two `when`? I guess it will decrease the level.

Reply:
Yes, done

Reply:
Thanks :)